### ISSUE TITLE
[Bug report]: Margin between text and icon in button not always applied

### ISSUE THREAD

**Issue by scania-mharlin** [scania-digital-design-system/tegel#589]
### Requirements before reporting

- [X] No duplicated [issue](https://github.com/scania-digital-design-system/tegel/issues) reported.
- [X] I have checked the latest version if the bug exist there. See all available packages at [npmJS.com](https://www.npmjs.com/package/@scania/tegel)
- [X] I have followed the installation guide.

### Package versions

@scania/tegel-angular: ^1.7.0

### Browser

Chrome

### Framework

Angular

### Version

Angular 17

### Reproduction steps

1. Add a tds-button
2. Add a tds-icon through the slot="icon" 
3. Add a text attribute to the button. 
4. Reload the page showing the component 

### Code example
```
<tds-button type="button" variant="primary" size="sm" text="button text">
    <tds-icon slot="icon" name="plus"></tds-icon>
</tds-button>
```
### Screenshots

![Screenshot 2024-04-04 at 09 39 34](https://github.com/scania-digital-design-system/tegel/assets/166006750/bccf594f-7d00-47fc-a5d1-fafdfba3a39e)
![Screenshot 2024-04-04 at 09 39 09](https://github.com/scania-digital-design-system/tegel/assets/166006750/205bc201-203b-404a-825b-b45b5f18b2b1)

### Expected behaviour

The margin between text and icon should always be applied, now it seems that it triggers sometimes but not always. Not sure as to why but it seems like the class "tds-button :not(.only-icon).sm.sc-tds-button-s>[slot=icon]" is not applied when the margin is missing. 

**Working Workaround** : If I use the label slot for my button text it's working as expected. 

### Console errors

none

### Contact information

magnus.harlin@scania.com

**Comment by theJohnnyMe** [comment]
I looked into this. I could not replicate it in our demo page, however, I think the issue was in over-complicated styling. I have made a PR to simplify it a bit. Hopefully, it resolves this problem, let's see after the release.

**Comment by theJohnnyMe** [comment]
Fix was released in 1.12.1 version. 
Please try and reopen this ticket if the issue is still present. ;)
